cupy/core/fusion.py

```diff
@@ -524,21 +541,29 @@ def _should_use_min_scalar(in_args):
             max_scalar_kind = -1
             for arg in in_args:
                 kind = _kind_score[arg.dtype.kind]
-                if arg.const is None:
+                if isinstance(arg, _FusionVarArray):
                     max_array_kind = max(max_array_kind, kind)
-                else:
+                elif isinstance(arg, _FusionVarScalar):
                     max_scalar_kind = max(max_scalar_kind, kind)
+                else:
+                    assert False
             return (max_scalar_kind != -1 and
                     max_array_kind >= max_scalar_kind)
 
         def can_cast1(args, in_dtypes):
             for i in six.moves.range(nin):
-                if args[i].const is None:
-                    if not numpy.can_cast(args[i].dtype, in_dtypes[i]):
+                arg = args[i]
+                if isinstance(arg, _FusionVarArray):
+                    if not numpy.can_cast(arg.dtype, in_dtypes[i]):
                         return False
-                else:
-                    if not numpy.can_cast(args[i].const, in_dtypes[i]):
+                elif isinstance(arg, _FusionVarScalar):
+                    scalar_value = arg._var.const
+                    if scalar_value is None:
+                        scalar_value = arg.dtype.type(0)
```

Comment:
Is this safe?

Reply:
Actually, it is not safe.

Reply:
hmm, I have no idea.
Let's use this once.